README.md

```diff
@@ -0,0 +1,220 @@
+```
+
+### NPM
+
+```bash
+npm install uploadcare-widget --save
+```
+
+```javascript
+import uploadcare from 'uploadcare-widget'
+```
+
+### Other install methods
+
+The official Uploadcare Widget [documentation](https://uploadcare.com/documentation/widget/#install)
+has more install methods.
+
+## Usage
+
+Once you’re done with the install, there are
+two simple steps to take to actually use the widget.
+
+**Set your [public key](https://uploadcare.com/documentation/widget/#option-public-key)**.
+This can also sit in the `<head>` section,
+
+```html
+<script>
+  UPLOADCARE_PUBLIC_KEY = 'YOUR_PUBLIC_KEY';
+</script>
+```
+
+Your secret key is not required for the widget
+(it’s quite careless for your page to include any
+secret keys anyway.)
+
+**Insert widget element** into your form,
+
+```html
+<input type="hidden" role="uploadcare-uploader" name="my_file" />
+```
+
+By default, the library looks for inputs with the specified
+`role` attribute and places widgets there.
+Once a file is uploaded, this `<input>` gets a
+CDN link with a file UUID. Your server then
+receives this link instead of file content.
+
+We suggest placing the widget somewhere at the top of your form.
+Unlike regular inputs, our widget starts uploading files **immediately**
+after they get selected by a user, not on form submission.
+That way users can fill out the rest of your form while an
+upload is in progress. This can be a real time saver.
+
+### Usage with React
+
+```javascript
+import React, {Component} from 'react'
+import uploadcare from 'uploadcare-widget'
+
+class Uploader extends Component {
+
+  componentDidMount() {
+    uploadcare.start({publicKey: 'YOUR_PUBLIC_KEY'})
+  }
+
+  render() {
+    return (
+      <div>
+        <input type='text' role='uploadcare-uploader'/>
+      </div>
+    )
+  }
+}
+
+export default Uploader
+```
+
+See [full demo](https://github.com/uploadcare/uploadcare-widget-react-demo/).
+
+### Usage with Angular
+
+We have the Angular 1 directive for Uploadcare Widget.
+See [angular-uploadcare](https://github.com/uploadcare/angular-uploadcare).
+
+For Angular 2, [this demo](https://plnkr.co/edit/6caWQ6cct4L3715LehxZ?p=preview) might be useful.
+
+## Configuration
+
+The widget is highly customizable through widget options.
+All configuration options together with ways to set them are
+described in [our docs](https://uploadcare.com/documentation/widget/#configuration).
+
+## JavaScript API
+
+You might not want to use all the features that
+[our widget](https://uploadcare.com/documentation/widget/) exhibits.
+Or, perhaps, you might want to redesign the user experience
+without having to reinvent the wheel.
+Maybe, you're in pursuit of building a UI on top of the widget.
+For all of those use cases, we provide a
+[JavaScript API](https://uploadcare.com/documentation/javascript_api/).
+Feel free to control the default widget with it,
+or make use of its standalone components that
+can be combined with your own solutions.
```

Comment:
I think I've already mentioned I don't like this paragraph, but we can still fix this later.